app/models/task.rb

```diff
@@ -0,0 +1,13 @@
+class Task < ActiveRecord::Base
+    def is_complete?
+        return completed_at != nil
+    end
+
+    def action=(val)
+        self[:action] = val.upcase
```

Comment:
This is a good thing to think about! The thing that would make the most sense for this, we haven't yet learned about. It is called an ActiveRecord callback, and it allows us to create a method that would be called before the ActiveRecord item was saved to the database. If you'd like to read about this more, here is a [link](http://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html)